apps/application-system/api/src/app/modules/application/utils/tokenUtils.ts

```diff
@@ -1,16 +1,32 @@
-import jwt from 'jsonwebtoken'
+import jwt, { VerifyErrors } from 'jsonwebtoken'
+
+import { logger } from '@island.is/logging'
 
 import { environment } from '../../../../environments'
 
-export function verifyToken<T>(token: string): T | null {
+export function verifyToken<T>(token: string): T | 'TokenExpiredError' | null {
   try {
-    const decoded = (jwt.verify(
+    return (jwt.verify(
       token,
       environment.templateApi.jwtSecret,
     ) as unknown) as T
+  } catch (e) {
+    const errors = e as VerifyErrors | null
+
+    if (errors?.name === 'TokenExpiredError') {
+      return 'TokenExpiredError'
+    }
 
-    return decoded
+    return null
```

Comment:
Not a fan of swallowing errors, losing information and returning a union type. It's better to throw error objects.

Maybe in this case you could extend the error with the decoded token (though in that case we might need to log and swallow a decode error).

```
type TokenError<T> = VerifyErrors & { decoded?: T }

// in verifyToken
catch (e) {
  const verifyError = e as TokenError<T>
  try {
    verifyError.decoded = jwt.decode(token) as unknown as T
  } catch (decodeError) {
    logger.warn('Cannot decode token', e)
  }
  throw verifyError
}
```

I don't know 🤷‍♂️ It's at least good to somewhere log the name/message of the jwt verify error.